Makefile

```diff
@@ -19,7 +21,26 @@ machine-controller: $(shell find cmd pkg -name '*.go') vendor
 				github.com/kubermatic/machine-controller/cmd/controller
 
 docker-image: machine-controller
+	make docker-image-nodep
+
+docker-image-nodep:
 	docker build -t $(IMAGE_NAME) .
+	if git describe --tags $(shell git rev-parse HEAD)|grep -v -- '-g'; then \
+		$(eval IMAGE_TAG = $(shell git describe --abbrev=0 --tags)) \
```

Comment:
So there are 2 things for me here:
- I would rather have 1 task called `push` which builds the image and pushes it. - Why have 2 tasks for it?
- I would rather use an environment variable for the tag. Parsing some output always makes me cringe

Reply:
I guess we could merge the targets, but I dislike relying on CI-Implementiation-Specific env vars

Reply:
Regarding the env var, i use TAG or something within the makefile and in the pipeline i would set ENV=CIRCLE_SHA1

Reply:
Done that